### PR TITLE
Search accessibility improvements

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

- search dialog now closes when user hits escape (sadly this does not work if the user has the Google search results iframe in focus)
- search toggle enhances to a button instead of link
- establish link between search toggle and search dialog
- search dialog marked up as dialog (and removed if search switch is off)
- on opening, search dialog receives focus to notify screen reader users what has happened
- on closing, search toggle receives focus

## Screenshots

aug-01-2018 19-41-21

## What is the value of this and can you measure success?

- Addresses some concerns raised in RNIB accessibility audit
- Makes our site more accessible to screen reader and keyboard users

### Accessibility test checklist

- [x] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [x] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)

### Tested

- [x] Locally

### CODE REVIEW

**Review by AWare — COMMENTED** [review]
looks like a pretty big improvement :)

**Review by AWare — APPROVED** [review]

**Comment by Ap0c** [comment]
Nice! 🚀

**Review by svillafe — APPROVED** [review]
🏃 ✅

File: static/src/javascripts/projects/common/modules/navigation/search.js [hunk 1]
```diff
@@ -97,11 +125,16 @@ class Search {
...
+                                            popup.focus();
                                             bean.on(
                                                 document,
                                                 'click',
                                                 maybeDismissSearchPopup
                                             );
+                                            document.addEventListener(
+                                                'keyup',
+                                                handleEsc
+                                            );
```
Comment (GHaberis) [inline static/src/javascripts/projects/common/modules/navigation/search.js]: I wonder if we should unbind these `click` and `keyup` event handlers once `dismissSearchPopup` has been called?

**Review by GHaberis — APPROVED** [review]
lovely 👌

**Pull request merged**